thirdeye-ui/src/app/components/admin-page/recent-failures/recent-failures.component.tsx

```diff
@@ -18,25 +18,73 @@ import TableCell from "@material-ui/core/TableCell";
 import { sortBy } from "lodash";
-import React, { FunctionComponent, useEffect, useMemo } from "react";
+import React, { FunctionComponent, useEffect, useMemo, useState } from "react";
 import { useTranslation } from "react-i18next";
+import { useSearchParams } from "react-router-dom";
 import { PageContentsCardV1, SkeletonV1 } from "../../../platform/components";
 import { ActionStatus } from "../../../rest/actions.interfaces";
 import { TaskStatus, TaskType } from "../../../rest/dto/taks.interface";
 import { useGetTasks } from "../../../rest/tasks/tasks.actions";
+import { WEEK_IN_MILLISECONDS } from "../../../utils/time/time.util";
 import { NoDataIndicator } from "../../no-data-indicator/no-data-indicator.component";
 import { LoadingErrorStateSwitch } from "../../page-states/loading-error-state-switch/loading-error-state-switch.component";
+import { TimeRangeQueryStringKey } from "../../time-range/time-range-provider/time-range-provider.interfaces";
+import { TimeRangeSelectorButton } from "../../time-range/v2/time-range-selector-button/time-range-selector-button.component";
 import { TaskRow } from "./task-row/task-row.component";
 
 export const RecentFailures: FunctionComponent = () => {
     const { t } = useTranslation();
     const { tasks, getTasks, status } = useGetTasks();
+    const [searchParams, setSearchParams] = useSearchParams();
 
-    useEffect(() => {
+    const [startTime, endTime] = useMemo(
+        () => [
+            Number(searchParams.get(TimeRangeQueryStringKey.START_TIME)),
+            Number(searchParams.get(TimeRangeQueryStringKey.END_TIME)),
+        ],
+        [searchParams]
+    );
+
+    const [selectedStart, setSelectedStart] = useState<number>(startTime);
+    const [selectedEnd, setSelectedEnd] = useState<number>(endTime);
+
+    const fetchTasks = (): void => {
         getTasks({
             status: [TaskStatus.TIMEOUT, TaskStatus.FAILED],
             type: [TaskType.DETECTION],
+            startTime: Number(startTime),
+            endTime: Number(endTime),
         });
+    };
+
+    useEffect(() => {
+        fetchTasks();
+    }, [startTime, endTime]);
+
+    const onHandleTimeRangeChange = (
+        startProp: number,
+        endProp: number
+    ): void => {
+        setSelectedStart(startProp);
+        setSelectedEnd(endProp);
+
+        searchParams.set(
+            TimeRangeQueryStringKey.START_TIME,
+            startProp.toString()
+        );
+        searchParams.set(TimeRangeQueryStringKey.END_TIME, endProp.toString());
+        setSearchParams(searchParams);
+    };
+
+    useEffect(() => {
+        // Apply the default filter of the last 7 days if no
+        // prior filter is specified or if the filters are invalid
+        if (!startTime || !endTime || startTime > endTime) {
+            const endRange = Date.now();
+            const startRange = endRange - WEEK_IN_MILLISECONDS;
+
+            onHandleTimeRangeChange(startRange, endRange);
+        }
     }, []);
```

Comment:
Typically I would let route redirection take care of setting the default params but this page is a little unique.

For this case, we can definitely simplify this by having the memoize function set the default if the expected parameters are not in the query params. I recommend moving the logic from lines `useEffect` in 79 - 88 to the `useMemo` function in 40-46.  You can remove the duplicate states in 48 and 49

Reply:
Trying to update the query params from `useMemo` does not work, and throws a console warning.


![Screenshot 2022-11-14 at 14 53 13](https://user-images.githubusercontent.com/20799363/201623368-e85a7611-764e-4ded-a38f-79fec64b1cd6.png)


![Screenshot 2022-11-14 at 14 51 29](https://user-images.githubusercontent.com/20799363/201622758-66de24a3-4bfb-4594-82bf-a622736b186e.png)

Reply:
Should I leave the query param updation on page load for page redirection to take care of later on? The reason for using the `useEffect` initially was to get around this issue.